Bottom: 23d40a75e6a519b98806f566d966a474845d86f5
Top:    6effe94308edae9f441d27bf85ed74ebdd86b085
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:57:50 +0000

polyon1.c: Rework InsCntr to re-use poly_Create() rather than open-coding

Removes the single case where the memory allocation macro MemGet was called, so
this was also removed.

This allows us to ensure common initialisation is performed in poly_Create().


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 4db2379..7b3e396 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1240,23 +1240,25 @@ InsCntr (jmp_buf * e, PLINE * c, POLYAREA ** dst)
 {
   POLYAREA *newp;
 
+  newp = poly_Create ();
+  if (newp == NULL)
+    error(err_no_memory);
+
   if (*dst == NULL)
     {
-      MemGet (*dst, POLYAREA);
-      (*dst)->f = (*dst)->b = *dst;
-      newp = *dst;
+      newp->f = newp->b = newp;
+      *dst = newp;
     }
   else
     {
-      MemGet (newp, POLYAREA);
       newp->f = *dst;
       newp->b = (*dst)->b;
       newp->f->b = newp->b->f = newp;
     }
+
+  c->next = NULL;
   newp->contours = c;
-  newp->contour_tree = r_create_tree (NULL, 0, 0);
   r_insert_entry (newp->contour_tree, (BoxType *) c, 0);
-  c->next = NULL;
 }				/* InsCntr */
 
 static void
